pallets/subtensor/tests/coinbase.rs

```diff
@@ -154,3 +155,1112 @@ fn test_set_and_get_hotkey_emission_tempo() {
+        let nominator2_stake =
+            SubtensorModule::get_stake_for_coldkey_and_hotkey(&nominator2, &hotkey);
+
+        log::debug!(
+            "Final stakes - Hotkey: {}, Nominator1: {}, Nominator2: {}, Total Hotkey Stake: {}",
+            delegate_stake,
+            nominator1_stake,
+            nominator2_stake,
+            total_hotkey_stake
+        );
+
+        // 9. Verify distribution
+        let min_take = SubtensorModule::get_min_delegate_take() as u64;
+        let total_emission = 20; // 10 per block for 2 blocks
+        let hotkey_emission = total_emission * min_take / u16::MAX as u64;
+        let remaining_emission = total_emission - hotkey_emission;
+
+        // We expect to distribute using the initial stake for nominator 1; because the delta is net positive
+        // We also use the INITIAL total hotkey stake
+        let nominator_1_emission =
+            remaining_emission * initial_nominator1_stake / intial_total_hotkey_stake;
+        let nominator_2_emission =
+            remaining_emission * initial_nominator2_stake / intial_total_hotkey_stake;
+
+        log::debug!(
+            "Calculated emissions - Hotkey: {}, Each Nominator: 1;{}, 2;{}",
+            hotkey_emission,
+            nominator_1_emission,
+            nominator_2_emission
+        );
+
+        // Debug: Print the actual stakes
+        log::debug!("Actual hotkey stake: {}", delegate_stake);
+        log::debug!("Actual nominator1 stake: {}", nominator1_stake);
+        log::debug!("Actual nominator2 stake: {}", nominator2_stake);
+
+        // Debug: Check the total stake for the hotkey
+        let total_stake = SubtensorModule::get_total_stake_for_hotkey(&hotkey);
+        log::debug!("Total stake for hotkey: {}", total_stake);
+
+        // Assertions
+        assert_eq!(delegate_stake, 2, "Hotkey stake mismatch");
+        assert_eq!(
+            nominator1_stake,
+            u64::try_from(
+                net_change
+                    .checked_add_unsigned(100 + nominator_1_emission as u128)
+                    .unwrap()
+            )
+            .unwrap(),
+            "Nominator1 stake mismatch"
+        );
+        assert_eq!(
+            nominator2_stake,
+            initial_nominator2_stake + nominator_2_emission,
+            "Nominator2 stake mismatch"
+        );
+
+        // 10. Check total stake
+        assert_eq!(
+            total_stake,
+            u64::try_from(
+                net_change
+                    .checked_add_unsigned(200 + total_emission as u128)
+                    .unwrap()
+            )
+            .unwrap(),
+            "Total stake mismatch"
+        );
+
+        log::debug!("Test completed");
+    });
+}
+
+// SKIP_WASM_BUILD=1 RUST_LOG=debug cargo test --package pallet-subtensor --test coinbase test_coinbase_nominator_drainage_with_net_negative_delta -- --nocapture
+#[test]
+fn test_coinbase_nominator_drainage_with_net_negative_delta() {
```

Comment:
Need to rethink this test.  doesn't appear to be correctly testing negative delta